File: sol/2195/2195.py
class Solution:
    def minimalKSum(self, nums: List[int], k: int) -> int:
        res = k * (k + 1) // 2
        nums = sorted(frozenset(nums))
        for x in nums:
            if x > k:
                break
            k += 1
            res += k - x
        return res

File: sol/2195/2195.cpp
struct Solution {
    using ll = long long;
    long long minimalKSum(vector<int>& nums, int k) {
        ll res = static_cast<ll>(k) * (k + 1) / 2;
        ranges::sort(nums);
        int n = nums.size();
        for (int i = 0; i < n; ++i){
            if (i != 0 && nums[i] == nums[i - 1]){
                continue;
            }
            if (nums[i] > k){
                break;
            }
            res += (++k) - nums[i];
        }
        return res;
    }
};
